packages/cloud_firestore/cloud_firestore/lib/src/query.dart

```diff
@@ -715,9 +723,12 @@ class _JsonQuery implements Query<Map<String, dynamic>> {
       if (operator == '!=') {
         assert(!hasNotEqualTo, "You cannot use '!=' filters more than once.");
         assert(!hasNotIn, "You cannot use '!=' filters with 'not-in' filters.");
-        assert(!hasDocumentIdField,
-            "You cannot use a FieldPath.documentId field with '!=' filters.");
+
         hasNotEqualTo = true;
+
+        if (field != FieldPath.documentId) {
+          hasNotEqualToOperatorAndNotDocumentIdField = true;
```

Comment:
What is the purpose of this assignment? From this point, the variable is not used anymore

Reply:
Wait ignore me, there's an hidden `for` loop